src/rt/test/perf/backpressure1/backpressure1.cc

```diff
@@ -0,0 +1,193 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+
+/**
+ * This tests a simple scenario for backpressure where many individual `Sender`
+ * cowns send messages to a single set of `Receiver` cowns. The `Recevier` cowns
+ * may be placed behind a chain of `Proxy` cowns to test backpressure
+ * propagation.
+ **/
+
+#include "test/log.h"
+#include "test/opt.h"
+#include "verona.h"
+
+#include <chrono>
+
+using namespace verona::rt;
+using timer = std::chrono::high_resolution_clock;
+
+struct Receiver;
+struct Proxy;
+static std::vector<Receiver*> receiver_set;
+static std::vector<Proxy*> proxy_chain;
+
+struct Receiver : public VCown<Receiver>
+{
+  static constexpr size_t report_count = 1'000'000;
+  size_t msgs = 0;
+  timer::time_point prev = timer::now();
+};
+
+struct Receive : public VAction<Receive>
+{
+  void f()
+  {
+    auto& r = *receiver_set[0];
+    r.msgs++;
+    if ((r.msgs % Receiver::report_count) != 0)
+      return;
+
+    const auto now = timer::now();
+    const auto t =
+      std::chrono::duration_cast<std::chrono::milliseconds>(now - r.prev);
+    logger::cout() << Receiver::report_count << " messages received in "
+                   << t.count() << "ms" << std::endl;
+    r.prev = now;
+  }
+};
+
+struct Proxy : public VCown<Proxy>
+{
+  size_t index;
+
+  Proxy(size_t index_) : index(index_) {}
+
+  void trace(ObjectStack& st) const
+  {
+    if (this != proxy_chain.back())
+    {
+      st.push(proxy_chain[index + 1]);
```

Comment:
This looks dodgy. What if index is the last one?

Reply:
If that were the case, then `this == proxy_chain.back()` would be true

Reply:
d'oh